### PR TITLE
Allow to show and edit tags in listings queries/mutations

### PR DESCRIPTION
**Pull request by joaodubas**
The last step of the tagging system is to be available in listings, so the following changes were made:

1. Listing type has a `tags` node:
   ```graphql
   query {
     listings {
       listings {
         id
         tags {
           nameSlug
           category
           visibility
         }
       }
     }
   }
   ```
2. Listing input type has a `tags` uuid list:
   ```graphql
   mutation {
     insertListing(input: {
       type: "Apartamento",
       address: {
         state: "SP",
         city: "São Paulo",
         neighborhood: "Vila Pompéia",
         postalCode: "05027-000",
         street: "Rua Doutor Miranda de Azevedo",
         streetNumber: "189",
         lat: -23.5308362,
         lng: -46.6909294
       },
       tags: [
         "3157962a-deac-46e1-a74f-9c9c45e50baf",
         "bc7317e6-480d-4ae4-a16a-ff94e4461499",
         "31ab8cd1-cfe7-4517-af26-7f693141c32d"
       ]
     }) {
       id
       tags {
         nameSlug
       }
     }
   }
   ```

### CODE REVIEW

File: apps/re/lib/listings/listings.ex [hunk 1]
```diff
@@ -234,6 +234,8 @@ defmodule Re.Listings do
     |> Repo.all()
   end
 
+  def upsert_tags(listing, nil), do: {:ok, listing}
```
Comment (rhnonose) [inline apps/re/lib/listings/listings.ex]: Should also handle empty list

File: apps/re_web/lib/graphql/types/listing.ex [hunk 2]
```diff
@@ -108,6 +110,8 @@ defmodule ReWeb.Types.Listing do
     field :address_id, :id
 
     field :development_uuid, :uuid
+
+    field :tags, list_of(:uuid)
```
Comment (rhnonose) [inline apps/re_web/lib/graphql/types/listing.ex]: This should be `non_null(list_of(non_null(:uuid)))` and a covering test for those cases.

File: apps/re/lib/listings/listings.ex (hunk 1 above)
Comment (joaodubas) [reply apps/re/lib/listings/listings.ex]: An empty list should remove associated tags (https://github.com/emcasa/backend/blob/8edd15ba6bcf56b453dbcbc27e35e09ac4196166/apps/re/test/listings/listings_test.exs#L530-L558)

File: apps/re_web/lib/graphql/types/listing.ex (hunk 2 above)
Comment (joaodubas) [reply apps/re_web/lib/graphql/types/listing.ex]: Tags can be null (maybe the user don't know what to insert, or the interface is only available for admin users at the launch), and the list can be empty, this way, we can remove all tags from a listing.

Comment (joaodubas) [reply apps/re_web/lib/graphql/types/listing.ex]: Besides that, both cases are covered in tests for `Re.Listings.upsert_tags/2`
https://github.com/emcasa/backend/blob/8edd15ba6bcf56b453dbcbc27e35e09ac4196166/apps/re/test/listings/listings_test.exs#L530-L558

File: apps/re/test/listings/listings_test.exs [hunk 3]
```diff
@@ -536,17 +533,29 @@ defmodule Re.ListingsTest do
...
+      listing = insert(:listing, user: user, tags: [tag_1, tag_2])
 
       {:ok, updated_listing} = Listings.upsert_tags(listing, [])
 
       assert Enum.count(updated_listing.tags) == 0
       refute Enum.member?(updated_listing.tags, tag_1)
       refute Enum.member?(updated_listing.tags, tag_2)
     end
+
+    test "noop when no tags is nil" do
```
Comment (pmargreff) [inline apps/re/test/listings/listings_test.exs]: Is the test description right? I guess it should be "shouldn't update if receives nil" or something like that ...

File: apps/re_web/lib/graphql/types/listing.ex (hunk 2 above)
Comment (rhnonose) [reply apps/re_web/lib/graphql/types/listing.ex]: The empty list makes sense, but being able to pass `["someuuid", nil]` doesn't.
So, it should be `list_of(non_null(:uuid))`

**Comment by rhnonose** [comment]
I think that's what I got confused in the empty array comment @pmargreff 
Creating/deleting the `listing <-> tag` relationship sounds better, but it's too SQLy, but having a add/remove mutation could be more explicit.

**Comment by joaodubas** [comment]
I think that having two mutations to handle associate/disassociate tag and listing can be cumbersome for the client, but it's a safer and explicit, will work on this in another PR.

**Comment by rhnonose** [comment]
Yeah, we can discuss it later, it's fine for this PR.

**Pull request merged**